sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAzureNamedKeyCredentialPolicy.java

```diff
@@ -131,7 +132,9 @@ private String getCanonicalizedResource(URL requestUrl) {
                 String queryParamValuesStr = String.join(",", queryParamValues);
 
                 if ("comp".equalsIgnoreCase(queryParamName)) {
```

Comment:
Is `comp` the only query parameter that will be part of the canonicalized resource? If so, can a lot of the sorting and looping logic be removed to just get the `comp` query parameter?

Reply:
This code was written a while ago before my time in Tables based on what Storage does, so I hadn't really take a good look at this class. Now that you mention it, it looks like we can definitely do that.

Reply:
Looks like the logic is correct

https://docs.microsoft.com/en-us/rest/api/storageservices/authorize-with-shared-key#shared-key-lite-and-table-service-format-for-2009-09-19-and-later